internal/controller/changetransferpolicy_controller.go

```diff
@@ -342,6 +343,15 @@ func (r *ChangeTransferPolicyReconciler) setCommitStatusState(ctx context.Contex
 			"toManyMatchingSha", tooManyMatchingShas,
 			"foundCount", len(csList.Items))
 	}
+
+	// Keep the URL from previous reconciliation where the phase was a success, if the commit status was not found, likely due to a sha mismatch.
+	for _, ctpStatusState := range targetCommitBranchState.CommitStatuses {
+		for i, calculatedCSState := range commitStatusesState {
+			if calculatedCSState.Key == ctpStatusState.Key && ctpStatusState.Url != "" {
+				commitStatusesState[i].Url = ctpStatusState.Url
+			}
+		}
+	}
```

Comment:
Hm, not loving this... if we want historical URLs, we should just have that in the status history feature.